src/induction/base.ml

```diff
@@ -193,18 +223,26 @@ let rec next (input_sys, aparam, trans, solver, k, unknowns) =
     (* Asserting implications if k > 0. *)
     if Numeral.(k > zero) then
       nu_unknowns
+      (* If we're skipping steps, we don't need to assert that the property
+         held in the previous step *)
+      |> List.filter (fun (name, _) -> match TransSys.get_prop_kind trans name with
+        | Reachable Some (From b)
+        | Reachable Some (At b) -> b < k_int
+        | Reachable Some (FromWithin (b, _))-> b < k_int
+        | _ -> true
+      ) 
       |> List.map (fun (_, term) -> Term.bump_state Numeral.(k-one) term)
       |> Term.mk_and |> SMTSolver.assert_term solver ;
 
     (* Filtering properties which are not known to be k-true at this step. *)
     let unknowns_at_k, k_true =
       nu_unknowns |> List.partition (
         fun (name,_) -> match TransSys.get_prop_status trans name with
-        | Property.PropKTrue k -> k_int > k
+        | Property.PropKTrue k -> k_int > k 
         | _ -> true
       )
     in
-
+    let k = if not no_skip then skip_steps_next trans solver k else k in
```

Comment:
This should be moved to line 216. After updating transition system but before filtering properties.